h/static/styles/partials/_base.scss

```diff
@@ -1,7 +1,23 @@
+/**
+ * 1. Necessary to fix IE floating footer bug
+ * TODO: Transition to CSS grid for page layout,
+ *       with flex reserved one-dimensional component layout
+ */
+
+html {
+  display: flex;
+}
+
 body {
   display: flex;
   flex-direction: column;
   justify-content: flex-start;
+  min-height: 100vh; /* 1 */
```

Comment:
Can you clarify what the `min-height` and `width` are needed for? On master I see that `min-height` is currently set to `100%` for the body.

Reply:
Specifying `min-height: 100vh;` vs 100% isn't strictly required (at least on the `<body>` tag), it's just more explicit, because it answers the question "100% of what?"

Width is needed because part of the solution is to set `display: flex;` on the `<body>`, and flex only lays out content in a single direction. We're using `flex-direction: column;` to get the behavior we want (footer all the way at the bottom), so the `<body>` flex layout doesn't care about horizontal alignment. If you inspect the page and turn off the `<body>` width declaration, at a large enough browser width, you should see the sidebar collapse inward towards the center of the page. Telling the `<body>` to be the width of the viewport fixes that. Unfortunately `min-width` causes a horizontal scrollbar when there's a vertical scrollbar, so `width` it is.

This seems like an opportune time to plug CSS grid once more... :wink: